sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -291,6 +293,75 @@ def read_item(  # pylint:disable=docstring-missing-param
         request_options["containerRID"] = self.__get_client_container_caches()[self.container_link]["_rid"]
         return self.client_connection.ReadItem(document_link=doc_link, options=request_options, **kwargs)
 
+    @distributed_trace
+    def read_items(
+            self,
+            items: Sequence[Tuple[str, _PartitionKeyType]],
+            *,
+            executor: Optional[ThreadPoolExecutor] = None,
+            max_concurrency: int = 10,
+            consistency_level: Optional[str] = None,
+            session_token: Optional[str] = None,
+            initial_headers: Optional[Dict[str, str]] = None,
+            excluded_locations: Optional[List[str]] = None,
+            **kwargs: Any
```

Comment:
The other methods have response_hook as an explicit keyword argument - is there a reason we chose not to do that?
```suggestion
            response_hook: Optional[Callable[[Mapping[str, str], Dict[str, Any]], None]] = None,
            **kwargs: Any
```

Reply:
@annatisch it is an entirely optional parameter that does not change behavior in any way and is not commonly used. truthfully it should be a kwarg everywhere, but it was initially documented in the other methods.

Reply:
Yeah - it's not a big problem and isn't blocking. Though being "optional" isn't really the concern - it's more that without that method signature declaration, a customer gets no IDE experience, and no static analysis for validation and maintenance.

We're actively looking for a way in the Azure SDK more broadly to lift _all_ pipeline-related optional keyword arguments to be fully typed - likely using some combination of the `Unpack` syntax.
And in fact - maybe that's an approach you could look at across your APIs - if you want to hide it somewhat as an advanced scenario, but still give customers a solid experience, sticking it a TypedDict with Unpack may work.

Either way it's not an issue that needs to be solved today, and is not blocking to this PR.

Reply:
ok I am resolving this comment for now and we can revisit it later.